model/task/task.go

```diff
@@ -322,6 +322,10 @@ type Task struct {
 	// NumNextTaskDispatches is the number of times the task has been dispatched to run on a
 	// host or in a container. This is used to determine if the task seems to be stuck.
 	NumNextTaskDispatches int `bson:"num_next_task_dispatches" json:"num_next_task_dispatches"`
+
+	// ProjectStorageMethod describes how the parser project for this task's version is
+	// stored. If this is empty, the default storage method is StorageMethodDB.
+	ProjectStorageMethod evergreen.ParserProjectStorageMethod `bson:"project_storage_method" json:"project_storage_method,omitempty"`
```

Comment:
Can we mention that this, kind of like the distro document on hosts, is the cached value and is not kept up-to-date and/or potentially name it `CachedProjectStorageMethod`? It only takes a snapshot of the storage method at the time the task is created, and isn't updated if the version's parser project switches to using S3 later on. For example, if the task is initially scheduled as part of the patch and starts out as having the parser project in the DB, and then later on, generate.tasks runs and moves the parser project to S3, this will still have the old project storage method (db).